module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,19 @@
+#include "sort.hpp"
+#include <algorithm>
+
+std::deque<std::string> lengthSort(std::forward_list<std::string>& forwardList) {
+    std::deque<std::string> dq;
+    
+    forwardList.sort( [](const auto str1, const auto str2){
+        if(str1.size() == str2.size()){
+            return str1 < str2;
+        }
+        return str1.size() < str2.size();
+    } );
+    
+    for(const auto el : forwardList){
+        dq.push_back(el);
+    }
```

Comment:
You can make it simpler (in one line) just by creating deque, sth similar to this: `std::deque<std::string> result_deque{input_list.begin(), input_list.end()};`

Reply:
As Kasperice said you can initialize deque using existing list. No point for push_back for

Reply:
+1